lib/matcher.js

```diff
@@ -420,6 +420,63 @@ export const canApplyRule = (rule, node) =>
     !!match(rule.matchFn, node, rule.constraints)
 
 /**
+ * Given an inputAST and a node to match, return an array with
+ * the path from the AST to the node. Accounts for multiple paths
+ *
+ * @param {input} input AST
+ * @param {node} subtree of input to be matched, TODO: handle when node is an array
+ * @param {path} an array of steps to be outputted
+ * @param {allPathsFound} an array containing duplicate paths if any
+ */
+
+export const getPath = (input, node, path = [], allPathsFound = []) => {
+    if (print(input) === print(node)) {
+        allPathsFound.push(path)
+    } else {
+        for (var i in input.args){
+            getPath(input.args[i]
+                         , node
+                         , path.concat(['args', i])
+                         , allPathsFound)
+        }
+    }
+    return allPathsFound
+}
+
+/**
+ *
+ * Given an inputAST and a path, return the node at the path
+ * @param {input} input AST
+ * @param {path} array containing steps to a node in the input AST
+ */
+export const getNodeAtPath = (input, path) => {
```

Comment:
I think this could be more succinctly implemented as a recursive function, e.g.:
```
const getNodeAtPath = (ast, path) => {
   if (!ast) {
       throw new Error("node doesn't exist at path");
   } else if (path.length > 0) {
       const [key, ...restOfPath] = path;
       return getNodeAtPath(ast[key], restOfPath);
   } else {
       return ast;
   }
}
```